sunkit_pyvista/plotter.py

```diff
@@ -584,3 +586,167 @@ def plot_limb(self, m, *, radius=0.02, **kwargs):
+            'bottom', 'top', 'left', 'right', 'front', or 'back'.
+            Default is 'bottom'.
+        component : int, optional
+            The component of the vector field to be plotted.
+            0, 1, or 2 for x, y, or z component, respectively.
+            Default is 2.
+        cmap : str, optional
+            The colormap for the boundary.
+            Default is 'gray'.
+        kwargs : dict
+            Keyword arguments for `pyvista.Plotter.add_mesh`.
+        """
+        self._create_mesh()
+        nx, ny, nz = self.mesh.dimensions
+        x_min, y_min, z_min = 0, 0, 0
+        x_max, y_max, z_max = nx - 1, ny - 1, nz - 1
+
+        if boundary == "bottom":
+            subset = (x_min, x_max, y_min, y_max, z_min, z_min)
+        elif boundary == "top":
+            subset = (x_min, x_max, y_min, y_max, z_max, z_max)
+        elif boundary == "left":
+            subset = (x_min, x_min, y_min, y_max, z_min, z_max)
+        elif boundary == "right":
+            subset = (x_max, x_max, y_min, y_max, z_min, z_max)
+        elif boundary == "front":
+            subset = (x_min, x_max, y_min, y_min, z_min, z_max)
+        elif boundary == "back":
+            subset = (x_min, x_max, y_max, y_max, z_min, z_max)
+        else:
+            raise ValueError("Invalid boundary. Choose from 'bottom', 'top', 'left', 'right', 'front', or 'back'.")
+
+        surface = self.mesh.extract_subset(subset).extract_surface()
+        surface.active_vectors_name = "vectors"
+        surface.active_scalars_name = "magnitudes"
+        self.add_mesh(surface, scalars="vectors", component=component, cmap=cmap, lighting=False, **kwargs)
+
+    def plot_field_lines(
+        self,
+        seeds,
+        render_lines_as_tubes=True,
+        radius=1,
+        max_steps=10000,
+        step_size=0.1,
+        seeds_config=dict(show_seeds=False, color="red", point_size=5),
+        **kwargs,
+    ):
+        """
+        Plot field lines traced from seeds using `streamtracer.StreamTracer`.
+
+        Parameters
+        ----------
+        seeds : ndarray
+            A (N, 3) array representing the seeds.
+        render_lines_as_tubes : bool, optional
+            Whether to render field lines as tubes.
+            Default is True.
+        radius : float, optional
+            The radius of the tubes for rendering field lines.
+            Default is 1.
+        max_steps : int, optional
+            The maximum number of steps for tracing field lines.
+            Default is 10000.
+        step_size : float, optional
+            The step size for tracing field lines.
+            Default is 0.1.
+        seeds_config : dict, optional
+            Configuration for plotting seeds.
+            Default is dict(show_seeds=False, color='red', point_size=5).
+        kwargs : dict
+            Keyword arguments for `pyvista.Plotter.add_mesh`.
+        """
+        tracer = StreamTracer(max_steps, step_size)
```

Comment:
VTK / pyvista has it's own streamline functionality which is probably a more natural fit for this package. Are you using streamtracer for some specific reason?

Reply:
I don't think the dependency is a big issue, it's just the cost of bouncing around between VTK and back. I'm happy to merge this like this, maybe we could open an issue though?